src/person.cpp

```diff
@@ -1660,7 +1675,89 @@ AnimateResult Person::OnAnimate(int delay)
 		return OAR_OK;
 	}
 
-	NOT_REACHED(); // We are truly lost now.
+	/* The person is walking at random over pathless land. */
+	this->DecideMoveDirectionOnPathlessLand(former_voxel, former_vox_pos, exit_edge, dx, dy, dz);
+	return OAR_OK;
+}
+
+/**
+ * The person is walking on pathless land, decide whether the current movement is allowed and where to go next.
+ * @param former_voxel The voxel from which the person comes.
+ * @param former_vox_pos The coordinates from which the person comes.
+ * @param exit_edge The edge over which the person left the voxel.
+ * @param dx The X coordinate change between the initial and the current position.
+ * @param dy The Y coordinate change between the initial and the current position.
+ * @param dz The Z coordinate change between the initial and the current position.
+ * @pre The person is located in (but not yet added to) the destination voxel, or somewhere below or above it.
+ */
+void Person::DecideMoveDirectionOnPathlessLand(Voxel *former_voxel,
+                                               const XYZPoint16 &former_vox_pos,
+                                               const TileEdge exit_edge,
+                                               const int dx,
+                                               const int dy,
+                                               const int dz)
+{
+	const XYZPoint16 init_pos = this->vox_pos;
+	Voxel *new_voxel = _world.GetCreateVoxel(this->vox_pos, true);
+	this->AddSelf(new_voxel);
+	this->UpdateZPosition();
+	new_voxel = _world.GetCreateVoxel(this->vox_pos, false);
+
+	assert(former_voxel->GetGroundType() != GTP_INVALID);
+	assert(new_voxel != nullptr && new_voxel->GetGroundType() != GTP_INVALID);
+	TileSlope old_voxel_slope = ExpandTileSlope(former_voxel->GetGroundSlope());
+	TileSlope new_voxel_slope = ExpandTileSlope(   new_voxel->GetGroundSlope());
+	/* Convert the bottom part of steep slopes to a format that is easier to handle here. */
+	auto convert_slope = [](TileSlope &slope) {
+		if ((slope & TSB_STEEP) == 0 || (slope & TSB_TOP) != 0) return;
+		const uint8 raised_corner_bit = (slope & ~TSB_STEEP);
+		slope = SL_FLAT;
+		if (raised_corner_bit != TSB_NORTH) slope |= TSB_SOUTH;
+		if (raised_corner_bit != TSB_SOUTH) slope |= TSB_NORTH;
+		if (raised_corner_bit != TSB_WEST ) slope |= TSB_EAST;
+		if (raised_corner_bit != TSB_EAST ) slope |= TSB_WEST;
+	};
```

Comment:
Complicated calculations!

I think the 1715-1718 lines can be replaced by (modulo renamed variables which you may or may not want to use):
`raised_bottom_bits = (TSB_NORTH | TSB_EAST | TSB_SOUTH | TSB_WEST) ^ raised_top_bit;`
(set all bits, then clear the top-bit).

Bit twiddling is an art :)

Reply:
I reverted the change now as actually this introduced a logical error, because the corner we want to remove from the bitset is the one *opposite* the raised corner (e.g. for a North steep slope the result should be N|W|E).

Reply:
Assuming you mean "S|W|E":

(all bits set xor the top) means all bits set except the top-bit  (1 xor 1 = 0)

Reply:
oh, I see

`TSB_MASK_ALL ^ ROR(raised_corner_bit, 4, 2);`